test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/StringLengthAttributeAdapterTest.cs

```diff
@@ -15,7 +16,7 @@ public class StringLengthAttributeAdapterTest
     {
         [Fact]
         [ReplaceCulture]
-        public void GetClientValidationRules_WithMaxLength_ReturnsValidationParameters_Localize()
+        public void AddValidation_WithMaxLength_AddsAttributes_Localize()
```

Comment:
Should test the special cases in the `StringLengthAttributeAdapter` (`0` and `int.MaxValue`).